nav2_dwb_controller/dwb_core/src/dwb_local_planner.cpp

```diff
@@ -467,47 +458,62 @@ DWBLocalPlanner::transformGlobalPlan(
     costmap->getResolution() / 2.0;
-  double sq_dist_threshold = dist_threshold * dist_threshold;
+
 
   // If prune_plan is enabled (it is by default) then we want to restrict the
   // plan to distances within that range as well.
-  double sq_prune_dist = prune_distance_ * prune_distance_;
+  double prune_dist = prune_distance_;
 
   // Set the maximum distance we'll include points before getting to the part
   // of the path where the robot is located (the start of the plan). Basically,
   // these are the points the robot has already passed.
-  double sq_transform_start_threshold;
+  double transform_start_threshold;
   if (prune_plan_) {
-    sq_transform_start_threshold = std::min(sq_dist_threshold, sq_prune_dist);
+    transform_start_threshold = std::min(dist_threshold, prune_dist);
   } else {
-    sq_transform_start_threshold = sq_dist_threshold;
+    transform_start_threshold = dist_threshold;
   }
 
   // Set the maximum distance we'll include points after the part of the part of
   // the plan near the robot (the end of the plan). This determines the amount
   // of the plan passed on to the critics
-  double sq_transform_end_threshold;
+  double transform_end_threshold;
   if (shorten_transformed_plan_) {
-    sq_transform_end_threshold = std::min(sq_dist_threshold, sq_prune_dist);
+    transform_end_threshold = std::min(dist_threshold, prune_dist);
   } else {
-    sq_transform_end_threshold = sq_dist_threshold;
+    transform_end_threshold = dist_threshold;
   }
 
-  // Find the first pose in the plan that's less than sq_transform_start_threshold
+  // Find the first pose in the global plan that's further than prune distance
+  // from the robot using integrated distance
+  auto prune_point = nav2_util::geometry_utils::first_after_integrated_distance(
+    global_plan_.poses.begin(), global_plan_.poses.end(), prune_dist);
+
+  // Find the first pose in the plan (upto prune_point) that's less than transform_start_threshold
   // from the robot.
   auto transformation_begin = std::find_if(
-    begin(global_plan_.poses), end(global_plan_.poses),
+    begin(global_plan_.poses), prune_point,
     [&](const auto & global_plan_pose) {
-      return getSquareDistance(robot_pose.pose, global_plan_pose) < sq_transform_start_threshold;
+      return euclidean_distance(robot_pose.pose, global_plan_pose) < transform_start_threshold;
     });
 
-  // Find the first pose in the end of the plan that's further than sq_transform_end_threshold
-  // from the robot
-  auto transformation_end = std::find_if(
-    transformation_begin, end(global_plan_.poses),
-    [&](const auto & global_plan_pose) {
-      return getSquareDistance(robot_pose.pose, global_plan_pose) > sq_transform_end_threshold;
-    });
+  // Find the first pose in the end of the plan that's further than transform_end_threshold
+  // from the robot using integrated distance
+  auto transformation_end =
+    nav2_util::geometry_utils::first_after_integrated_distance(
```

Comment:
The RPP version of this logic uses `find_if` here: 

```
  auto transformation_end = std::find_if(
    transformation_begin, global_plan_.poses.end(),
    [&](const auto & pose) {
      return euclidean_distance(pose, robot_pose) > max_costmap_extent;
    });
```

I'm not really sure if it matters one way or another, but shouldn't we keep them consistent? I know this is from @sathak93 's PR so maybe he can let us know if there's a reason for the change here or if we can homologate them

Reply:
this was same as that in RPP in my PR. @Aposhian  changed it in  RPP now. i think it is better to make them consistent.